lib/devise-security/models/secure_validatable.rb

```diff
@@ -90,7 +90,9 @@ module ClassMethods
 
         def has_uniqueness_validation_of_login?
           validators.any? do |validator|
-            validator.kind_of?(ActiveRecord::Validations::UniquenessValidator) &&
+            defined? ActiveRecord && validator.kind_of?(ActiveRecord::Validations::UniquenessValidator) &&
+              validator.attributes.include?(login_attribute)
```

Comment:
You should have a `next` here, because it's possible that a project could have both `ActiveRecord` and `Mongoid` loaded.

Reply:
a cleaner version could be:

```
          validators.any? do |validator|
            validator_orm_klass = defined?(ActiveRecord) ? ActiveRecord::Validations::UniquenessValidator : ::Mongoid::Validatable::UniquenessValidator
            validator.kind_of?(validator_orm_klass) && validator.attributes.include?(login_attribute)
          end
```

but I think it shouldn't work in your example. Can you help me with this?

Reply:
I just mean that you need something like `next if validator.attributes.include?(login_attribute)`.

Reply:
The original code (upstream/master) is:
``````
def has_uniqueness_validation_of_login?
  validators.any? do |validator|
    validator.kind_of?(ActiveRecord::Validations::UniquenessValidator) &&
    validator.attributes.include?(login_attribute)
  end
end
``````

my new proposal is:
``````
def has_uniqueness_validation_of_login?
  validators.any? do |validator|
    validator_orm_klass = DEVISE_ORM == :active_record ? 
    ActiveRecord::Validations::UniquenessValidator : ::Mongoid::Validatable::UniquenessValidator
    validator.kind_of?(validator_orm_klass) && validator.attributes.include?(login_attribute)
  end
end
``````
DEVISE_ORM can have only one value at time.